application.go

```diff
@@ -240,42 +263,59 @@ func (a *Application) Run() error {
 		os.Exit(1)
 	}
 
-	defaultPlatformPlugin.glfwTasker = tasker.New()
-
-	m := newWindowManager()
-	m.forcedPixelRatio = a.config.forcePixelRatio
-
-	m.glfwRefreshCallback(a.window)
-	a.window.SetRefreshCallback(m.glfwRefreshCallback)
-	a.window.SetPosCallback(m.glfwPosCallback)
+	// Setup a new windowManager to handle windows pixel ratio's and pointer
+	// devices.
+	windowManager := newWindowManager(a.config.forcePixelRatio)
+	// force first refresh
+	windowManager.glfwRefreshCallback(a.window)
+	// Attach glfw window callbacks for refresh and position changes
+	a.window.SetRefreshCallback(windowManager.glfwRefreshCallback)
+	a.window.SetPosCallback(windowManager.glfwPosCallback)
 
-	// flutter's PlatformMessage handler is registered through the dart:ui.Window
-	// interface. ui.Window must have at least paint one frame, before any
-	// platfrom message can be corectly handled by ui.Window.onPlatformMessage.
-	glfw.WaitEvents()
+	// TODO: Can this only be done here? Why not in the plugin/glfwPlugin init loop above?
```

Comment:
When the plugin maker gets access to the texturer, (`flutter.TextureRegistry`).
He can create a new texture `myTexture := registry.NewTexture()`, and without waiting, call `myTexture.FrameAvailable()`.

When doing so, the engine must be running and ready to process texture.

Reply:
Moving the init loop here can be a possibility.

Reply:
I think the init loop for plugins must happen before the engine starts, so that channels are registered before Dart code starts sending messages to the Go side.

I'm assuming any use of the testurer API will be initiated by a call from the Dart side, so the texturer wont' be used before the engine is alive.
The normal plugin init also exposes the messenger, even though it can't be used yet because the engine isn't running. So far, all plugins I've seen don't do anything until asked via message/method call.

Reply:
> I think the init loop for plugins must happen before the engine starts, so that channels are registered before Dart code starts sending messages to the Go side.

Agree!
But, in our case it's more like `the init loop for plugins must happen before we start processing embedder.FlutterTask, so that channels are ready before the embedder starts handling platform messages to the Go side`

Platform messages aren't  processed until in the `eventLoop.WaitForEvents`, which occurs after  the plugin init loop.
```go
for !a.window.ShouldClose() {
	eventLoop.WaitForEvents(func(duration float64) { // <- Process Platform task
		glfw.WaitEventsTimeout(duration)
	})
	defaultPlatformPlugin.glfwTasker.ExecuteTasks()
	messenger.engineTasker.ExecuteTasks()
}
```
 
---

> I'm assuming any use of the testurer API will be initiated by a call from the Dart side, so the texturer wont' be used before the engine is alive.

While call to the [`handle`](https://github.com/go-flutter-desktop/examples/blob/b15e61712768e93b37934b8e68cf7975f2d0f5a9/texture_tutorial/go-texture-example/image/main.go#L46) of a `texture` is made from the engine side. Call from the Golang [`texture.FrameAvailable()`](https://github.com/go-flutter-desktop/examples/blob/b15e61712768e93b37934b8e68cf7975f2d0f5a9/texture_tutorial/go-texture-example/video/main.go#L53) to the engine can be made at any time, the engine must be running otherwise we get segfault.

Reply:
Plugin is also used internally (by the [`textinputPlugin`](https://github.com/go-flutter-desktop/go-flutter/blob/19ec300ac7e16fc0e4b0bc4d976a646136c5f404/textinput.go#L17) for example), the plugins init loop occurs before we set the GLFW handles, it's a non-issue.